linux/net/rina/dtcp.c

```diff
@@ -747,9 +1125,32 @@ static int rcv_flow_ctl(struct dtcp * dtcp,
         pci = pdu_pci_get_rw(pdu);
         ASSERT(pci);
 
-        snd_rt_wind_edge_set(dtcp, pci_control_new_rt_wind_edge(pci));
+        rt = pci_control_sndr_rate(pci);
+        tf = pci_control_time_frame(pci);
+
+        // Is this a window based control flow?
+        if(dtcp_window_based_fctrl(dtcp_config_get(dtcp))) {
+        	snd_rt_wind_edge_set(dtcp, pci_control_new_rt_wind_edge(pci));
+        	dt_sv_window_closed_set(dtcp->parent, false);
```

Comment:
I do not get it. The if clause is ok, so distinguishing between window or rate based fc. My question is, why in the case of window base you set ```dt_sv_window_closed_set(dtcp->parent, false);``` (line 1134)